src/org/mirah/jvm/mirrors/better_scope.mirah

```diff
@@ -601,6 +652,14 @@ class MethodScope < BetterScope
     @locals = Locals.new
     @imports = ImportsAndSearchPackages.new
   end
+  
+  def initialize(source: ClosureScope, context: Node)
+    super(context)
+    @scoper     = source.internal_scoper
+    @locals     = source.internal_locals
+    @imports    = source.internal_imports
```

Comment:
No, this is for converting a ClosureScope into a MethodScope when the closure is converted into a method, where the body of the closure becomes the body of the method. Hence, the old ClosureScope will not be reused but just discarded. And would happen only if the block does not contain methods at all.